packages/mux-video/src/index.ts

```diff
@@ -415,22 +426,28 @@ class MuxVideoElement extends CustomVideoElement<HTMLVideoElement> implements Pa
     super.attributeChangedCallback(attrName, oldValue, newValue);
 
     switch (attrName) {
+      case Attributes.ENV_KEY:
+      case Attributes.STREAM_TYPE:
+      case Attributes.DEBUG:
+      case Attributes.CUSTOM_DOMAIN:
+      case Attributes.BEACON_COLLECTION_DOMAIN:
+      case Attributes.EXPERIMENTAL_CMCD:
+      case Attributes.PREFER_CMCD:
+      case Attributes.DISABLE_COOKIES:
+        this.load();
+        break;
       case Attributes.PLAYER_SOFTWARE_NAME:
         this.playerSoftwareName = newValue ?? undefined;
+        this.load();
         break;
       case Attributes.PLAYER_SOFTWARE_VERSION:
         this.playerSoftwareVersion = newValue ?? undefined;
+        this.load();
         break;
       case 'src': {
-        const hadSrc = !!oldValue;
-        const hasSrc = !!newValue;
-        if (!hadSrc && hasSrc) {
-          this.load();
-        } else if (hadSrc && !hasSrc) {
-          this.unload();
-          /** @TODO Test this thoroughly (async?) and confirm unload() necessary (CJP) */
```

Comment:
because of this new async source loading behavior, we don't need this piece and can rely on unloading if `!newValue` and loading on a `newValue`?

Reply:
I might put this back, I over simplifed it

Reply:
right now it is fine because unload only has the core teardown which is done again in initialize() but there could be a case when the unload method holds more local logic so safer to put it back